rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+- I would like to generate metadata providing the experimental and scientific context about the cells in my matrix.
+- I would like to generate my matrix in a format that is compatible with standard single-cell analysis tools.
+- I would like to generate my matrix in a reasonable amount of time; in general, on the order of minutes, not hours.
+
+As a user with a pipette, I would like gene expression data for HCA projects.
+
+- To support this set of users, an interface via the HCA Data Browser will leverage the API to provide static matrix products.
+
+## Scientific "guardrails"
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+### Overview
+
+The Matrix Service enables on-the-fly expression matrix generation via a [RESTful API](https://matrix.data.humancellatlas.org/). Through user studies and iterative development over the past year, four major requirements of matrices have been identified and must be met:
+
+1. Matrices must enable cell-level granularity integrative analysis across data in the HCA.
+1. Matrices must include rich metadata that users are interested in so that they can better understand the data they're working with.
+1. Matrices must be compatible with downstream single cell analysis tools, for example in `scanpy` and `Seurat`, to further enable downstream analysis of HCA data by the wider community.
+1. Performance of serving matrices must be prioritized as the scale of HCA data is anticipated to reach or exceed the order of tens of millions of cells.
+
+The Matrix Service API is designed to fulfill these requirements in order to provide access to HCA data in a format that enables and encourages downstream integrative analysis.
+
+The default and recommended file format for matrices generated by the service is [loom](http://loompy.org/), an HDF5 file format that is optimized for storing large omics datasets such as expression matrices and designed for consumability, for example, by exporting metadata with matrices. The Matrix Service achieves its first three goals by delivering matrices in `loom`, and other supporting formats, via an expressive API that enables users the ability to specify individual cells and desired metadata to generate a matrix for. For a detailed example, see [this user vignette](https://nbviewer.jupyter.org/github/HumanCellAtlas/matrix-service/blob/master/docs/HCA%20Matrix%20Service%20to%20Scanpy.ipynb#HCA-Matrix-Service-Loom-Output) describing a `loom` matrix generated by the service. The fourth requirement is satisfied by a lazy cache of matrices that enables the availability of previously requested matrices within minutes, if not seconds.
+
+*Note: The current technical architecture of the service is documented [here](https://allspark.dev.data.humancellatlas.org/HumanCellAtlas/matrix-service/wikis/Technical-Architecture).*
+
```

Comment:
Is there a plan to put this documentation in a publicly accessible location. I don't have an allspark account and I can't read these documents?

Reply:
There isn't a plan to -  though, are you able to authenticate with GitHub? If not, I can see if we can get you access. Several DCP components host their developer documentation on GitLab (allspark)

Reply:
So I don't have access, I guess my github account has never been added to the HCA gitlab. I am concerned that a public document such as an RFC refers to private documentation. It would be wonderful if even a short summary was posted here or somewhere else publicly

@maniarathi it would be great to hear the thoughts from tech architecture about why this documentation is private and if there are good ways to expose any of it publicly